worker/uniter/runner/context/context.go

```diff
@@ -881,18 +882,46 @@ func (ctx *HookContext) GetSecret(uri *coresecrets.URI, label string, refresh, p
 	return v, nil
 }
 
-func (ctx *HookContext) getPendingSecretValue(uri *coresecrets.URI) (coresecrets.SecretValue, bool) {
-	if uri == nil {
+func (ctx *HookContext) getPendingSecretValue(uri *coresecrets.URI, label string, refresh, peek bool) (coresecrets.SecretValue, bool) {
+	if uri == nil && label == "" {
```

Comment:
Better to return an error?

Reply:
This was sort of existing code. Right before the method is called, there is a check
```
	if uri == nil && label == "" {
		return nil, errors.NotValidf("empty URI and label")
	}
```
So the error would be raised before the method is even called. The check is just to short circuit unnecessary logic. The method would work even without the check, but it would execute a couple of loops.